### PR TITLE
ES-81 : Added new OIDC Client create and update APIs through PMS

### PR DESCRIPTION
**Pull request by balaji-alluru**

### CODE REVIEW

File: partner/partner-management-service/src/main/java/io/mosip/pms/oidc/client/dto/ClientDetailCreateRequestV2.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class ClientDetailCreateRequestV2 extends ClientDetailCreateRequest{
+
+	 @NotEmpty
+	 private Map<@Size(min=3, max=3) String, @NotBlank String> clientNameLangMap;
```
Comment (LoganathanSekar7627) [inline partner/partner-management-service/src/main/java/io/mosip/pms/oidc/client/dto/ClientDetailCreateRequestV2.java:23]: Please handle varying size, if possible make it configurable.

**Review by LoganathanSekar7627 — CHANGES_REQUESTED** [review]

File: partner/partner-management-service/src/main/java/io/mosip/pms/oidc/client/dto/ClientDetailCreateRequestV2.java (hunk 1 above)
Comment (balaji-alluru) [reply partner/partner-management-service/src/main/java/io/mosip/pms/oidc/client/dto/ClientDetailCreateRequestV2.java:23]: we want to limit the key size to 3 as the key will be lang code which is always a 3 letter code.

File: partner/partner-management-service/src/main/java/io/mosip/pms/oauth/client/dto/CreateClientRequestV2Dto.java
Comment (ase-101) [inline partner/partner-management-service/src/main/java/io/mosip/pms/oauth/client/dto/CreateClientRequestV2Dto.java:1]: Lets follow a single naming convention, as above this should be renamed to CreateClientRequestDtoV2.java

File: partner/partner-management-service/src/main/java/io/mosip/pms/oauth/client/service/ClientManagementService.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    ClientDetail getClientDetails(String clientId);
+
+    /**
+     * API to create OIDC Client with new Client Name Lang Map 
+     * based on V2 OIDC Client Create API of Esignet
+     * @param createRequest
+     * @return
+     * @throws Exception
+     */
+	ClientDetailResponse createOAUTHClient(ClientDetailCreateRequestV2 createRequest) throws Exception;
```
Comment (ase-101) [inline partner/partner-management-service/src/main/java/io/mosip/pms/oauth/client/service/ClientManagementService.java:46]: lets rename the method to createOAuthClient

**Review by ase-101 — APPROVED** [review]

**Pull request merged**